WifiART/src/structs/Subelement.java

```diff
@@ -24,6 +24,7 @@
      * Produces the hex-buffer encoding for this LCI/LCR subelement
      *
      * @return The hex-buffer encoding
+     * @throws RuntimeException if the buffer cannot be calculated.
      */
-    String toHexBuffer();
+    String toHexBuffer() throws RuntimeException;
```

Comment:
The reason why I added it here is because an exception might be thrown in the toHexBuffer method, and I wanted the caller to be aware of this so the user can be notified. But it looks like RuntimeExceptions are unique in that interface implementations can throw them without the exception being specified in the interface's method signature. I will try throwing exceptions only in the implementation, and see if that works. In this case the implementation will throw something more specific like a NullPointerException.

Reply:
Yes if you would like the exception to be specific make it a checked exception, so the caller would have to handle it.